### PR TITLE
feat: add remote item id to search report responses

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Adds the remote item id to search `REPORT` responses for shared resources.

This is needed for the Web client to correctly resolve into a share from a search result. The Web client used the `shareId` in the past, however, with https://github.com/owncloud/web/issues/9784 we need to use the actual resource id (= remote item id) for this.

@dschmidt and I also thought about making the property more general by naming it `spaceId` and exposing it not only for shares but for all search results. But since we only need this for shares in this is the least invasive approach, we decided to go this way. Opinions welcome though.

## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/9094

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

### CODE REVIEW

**Review by dschmidt — APPROVED** [review]

File: services/search/pkg/search/service_test.go [hunk 1]
```diff
@@ -345,6 +352,11 @@ var _ = Describe("Searchprovider", func() {
 				Expect(match.Entity.Name).To(Equal("Shared.pdf"))
 				Expect(match.Entity.Ref.ResourceId.OpaqueId).To(Equal(mountpointSpace.Root.OpaqueId))
 				Expect(match.Entity.Ref.Path).To(Equal("./to/Shared.pdf"))
+				Expect(match.Entity.RemoteItemId).To(Equal(&searchmsg.ResourceID{
+					StorageId: grantSpace.RootInfo.Id.StorageId,
```
Comment (micbar) [inline services/search/pkg/search/service_test.go:356]: Please always use Getters from the protobuf generated types `GetRootInfo().GetID().GetStorageID()`

File: services/webdav/pkg/service/v0/search.go [hunk 2]
```diff
@@ -185,6 +185,11 @@ func matchToPropResponse(ctx context.Context, match *searchmsg.Match) (*propfind
 	if match.Entity.Ref.ResourceId.StorageId == utils.ShareStorageProviderID {
 		propstatOK.Prop = append(propstatOK.Prop, prop.Escaped("oc:shareid", match.Entity.Ref.ResourceId.OpaqueId))
 		propstatOK.Prop = append(propstatOK.Prop, prop.Escaped("oc:shareroot", match.Entity.ShareRootName))
+		propstatOK.Prop = append(propstatOK.Prop, prop.Escaped("oc:remote-item-id", storagespace.FormatResourceID(provider.ResourceId{
+			StorageId: match.Entity.RemoteItemId.StorageId,
```
Comment (micbar) [inline services/webdav/pkg/service/v0/search.go:189]: same

File: services/search/pkg/search/service_test.go (hunk 1 above)
Comment (JammingBen) [reply services/search/pkg/search/service_test.go:356]: But this is a test and `grantSpace` is a mock, so `GetRootInfo().GetID().GetStorageID()` seems to be unavailable on this mock. Do I need to add something to make this work? Other tests also don't use getters though 🤔

**Comment by JammingBen** [comment]
@micbar The Go vulnerability check fails now that I added https://github.com/owncloud/ocis/pull/9095/commits/f594db6067bca61f1f88e31f6e646d695b91e602. Could this be related? Other places in this webdav search service also don't use the protobuf getters 🤷

**Review by rhafer — COMMENTED** [review]
Somehow I wonder what the semantics of the `remote item id`  in the dav response are (and I also wonder about the naming :speak_no_evil: ). 

AFAIU the goal is to be able to reference where a search result is coming from, correct?

So if a search result is a file inside for a shared folder, you need to know the resource ID of the shared folder. Right?

If it's a file inside of a space that the user is able to access you need the resource id of the space root. Also right?

If that is true, should it really be called `remoteItemId`? I guess this is coming from the graph services `remoteItem` property in the sharedWithMe/sharedByMe responses? 

I'd appreciate some more details in the commit message about why this is needed and what the new property actually represents. (I fear I know the answer already, but do we document all the extended webdav properties  and their meaning somewhere?)

I can't really give a good review for this. As I don't know the search code enough. So this is neither a :+1:  or  a :-1: just some thoughts.

**Comment by JammingBen** [comment]
Thanks for your input!

> So if a search result is a file inside for a shared folder, you need to know the resource ID of the shared folder. Right?

Yes. 

> If it's a file inside of a space that the user is able to access you need the resource id of the space root. Also right?

Yes. The Web client extracts the space id from the file id. I know that's not ideal, but that's how it currently works.

> If that is true, should it really be called `remoteItemId`? I guess this is coming from the graph services `remoteItem` property in the sharedWithMe/sharedByMe responses?

Yes, the naming comes from the Graph terminology.

There is already a dav property called `shareRoot`, so we could also stay with that terminology and name the new property `shareRootId`, as this is the file id matching the share root path.

**Comment by JammingBen** [comment]
> I added the ones where i know that we use them. Which ones are missing?

@micbar These are missing at a first glance:

* `oc:shareid`
* `oc:shareroot`
* `oc:activelock`
* `oc:audio`
* `oc:location`
* `oc:public-link-item-type`
* `oc:public-link-permission`
* `oc:public-link-expiration`
* `oc:public-link-share-datetime`
* `oc:public-link-share-owner`
* `oc:trashbin-original-filename`
* `oc:trashbin-original-location`
* `oc:trashbin-delete-datetime`

How do we want to continue with this PR? I can make the commit message more descriptive and explain why and how to use this new property. Also, as stated above, I could rename it to `shareRootId`, to match with the naming convention of `shareRoot`. Though that might be confusing because it's different to the Graph naming terminology.

File: services/search/pkg/search/service_test.go (hunk 1 above)
Comment (micbar) [reply services/search/pkg/search/service_test.go:356]: That was just my „inner linter“ parsing the diff. Not an issue in tests.

**Comment by JammingBen** [comment]
@micbar As discussed, I added docs for the missing supported webdav props via https://github.com/owncloud/ocis/pull/9095/commits/783692efdc0942f4192ace97f52c3c005b686eaa and added more detail to the original commit message.

**Pull request merged**